### DISCUSSION TITLE
How to interact with date_picker.dart flutter library?

### DISCUSSION THREAD

**Discussion by proximilian** [leancodepl/patrol#1295] [Q&A]
Hi guys,
At the mobile app I am currently testing there is a date picker calendar, which is not developed by our devs but just used and actually it is a library that comes from Flutter itself -> material/date_picker.dart. At the widget tree there is only "Theme". Only the Widget Details Tree provide some info. Please advise, how to interact with it using Patrol. So far Patrol is doing excellent, but I have never interact with such "widgets". Is there some special approach? 
Thanks in advance for your help.
![Calendar](https://github.com/leancodepl/patrol/assets/125548662/b38ef03e-60db-4b72-a21c-3894f599cad7)

**Comment by bartekpacia** [comment]
Hi @proximilian, sorry for the delay.

Can you attach an example that I can clone and play with?

**Reply by proximilian** [reply]
Hi @bartekpacia,

Unfortunately our dev department was very busy these days and they had no time to provide an extract to attach. Meanwhile I spent some time on it and found that although only the Widget Details Tree can be used, the dates and all buttons can be interacted with only by the visual text e.g. `$(Theme).$('15').tap();` That seemed to be not the best but at least working option. Also found that if the automation testers want to be more flexible and to interact by some widget properties, they have to switch to and access the Widget tester - e.g.  `tester.tap(find.bySemanticsLabel('9, Tuesday, May 9, 2023'))` In that way the the finder can be considered as more reliable involving particular date details. The cool thing is that Patrol finders can be mixed with the widget tester ones.

**Reply by bartekpacia** [reply]
I'm glad you found a fix:)

Do you think `$('Some text')` should also match semantic labels?
